### PR TITLE
INTG-591: Incorporate the functionality of PagerDuty alerts when deployment fails to make to cloud-hub.

### PR DESCRIPTION
**Pull request by faisalkhanani**
…oyment fails to make.

INTG-591: Incorporate the functionality of PagerDuty alerts when deployment fails to make to cloud-hub.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -39,6 +40,9 @@ async function main() {
     catch (error) {
         console.error(error);
         core.setFailed(error.message);
+        if (PAGERDUTY_INTEGRATION_KEY) {
+            pager.makeAndSendPagerAlert(PAGERDUTY_INTEGRATION_KEY, error);
+        }
```
Comment (ryan-breidenbach) [inline src/index.js:45]: This error handling block is repeated several times. Does it makes sense to pull it into a helper function?

File: src/pagerduty.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                "custom_details": {
+                    "run_details": `https://github.com/${context.repo.owner}/${context.repo.repo}/actions/runs/${context.runId}`,
+                    "error_message": errorMessage,
+                    "related_commits": context.payload.commits
+                        ? context.payload.commits.map((commit) => `${commit.message}: ${commit.url}`).join(', ')
+                        : 'No related commits',
+                },
+            },
+                "routing_key": integrationKey,
+                "event_action": 'trigger'
```
Comment (ryan-breidenbach) [inline src/pagerduty.js:24]: I think these two lines should be unindented one level.

File: src/pagerduty.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+
+async function makeAndSendPagerAlert(integrationKey, errorMessage) {
+    try {
+        let alert = {
+            "payload": {
+                "summary": `${context.repo.repo}: Error in "${context.workflow}" run by @${context.actor}`,
+                "timestamp": new Date().toISOString(),
+                "source": 'MULESOFT-DEPLOY-ACTION',
+                "severity": 'critical',
```
Comment (ryan-breidenbach) [inline src/pagerduty.js:14]: Is this the intention - send a critical alert? If so, do we want this across all environments - DEV, UAT, and PROD?

**Comment by ryan-breidenbach** [comment]
I like the idea behind this so we can be more proactive about addressing failed builds.

File: src/pagerduty.js (hunk 3 above)
Comment (faisalkhanani) [reply src/pagerduty.js:14]: Actual severity depends on integration with **high urgency** or **low urgency** alerts setup in PD. We can manage this by passing different integration keys.

**Review by ryan-breidenbach — APPROVED** [review]

**Pull request merged**